internal/action/actions.go

```diff
@@ -1240,45 +1240,31 @@ func (h *BufPane) JumpLine() bool {
 // Start moves the viewport to the start of the buffer
 func (h *BufPane) Start() bool {
 	v := h.GetView()
-	v.StartLine = 0
+	v.StartLine = display.SLoc{0, 0}
 	h.SetView(v)
 	return true
 }
 
 // End moves the viewport to the end of the buffer
 func (h *BufPane) End() bool {
-	// TODO: softwrap problems?
 	v := h.GetView()
-	if v.Height > h.Buf.LinesNum() {
-		v.StartLine = 0
-		h.SetView(v)
-	} else {
-		v.StartLine = h.Buf.LinesNum() - v.Height
-		h.SetView(v)
-	}
+	v.StartLine = h.Scroll(h.SLocFromLoc(h.Buf.End()), -v.Height+1)
```

Comment:
The use of `v.Height` here (and in other functions in this file) is buggy, since the actual buffer height is `v.Height-1` if statusline is on. This is a separate issue.